akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2Compliance.scala

```diff
@@ -4,17 +4,39 @@
 
 package akka.http.impl.engine.http2
 
+import scala.annotation.switch
+
 /** INTERNAL API */
 private[akka] object Http2Compliance {
+  def parseSettungsEnablePushValue(incomingValue: Int): Boolean =
+    (incomingValue: @switch) match {
+      case 0 ⇒ false
+      case 1 ⇒ true
+      case _ ⇒ throw new IllegalEnablePushValueSettingException(s"Illegal value in SETTINGS_ENABLE_PUSH settings header! MUST BE 0 or 1, but was: [$incomingValue]")
+    }
+
+  private val MinOutMaxFrameSize = Math.pow(2, 14).toInt // minimum allowed to be set
```

Comment:
Maybe put these together with the other spec related constants into Http2Protocol.